### PR TITLE
Cleanup routing

### PR DESCRIPTION
**Pull request by andyneff**
I saw you removed the VPNKit `resolv.conf` file here: d1a3e64d2855764caba01f5c79d4c74dc90d1f6e and was wondering what was going on.

# TL;DR

If we are going to use the default `resolv.conf` file, I think we need to do a little more work on the routing table.

## My mess of a travel to your conclusion

The default `resolv.conf` does not currently work with this version of the script.

As I understand it:

image

(IPs are just as examples, they obviously vary)

## WSL1

In WSL1, the networks stack is one in the same with the host, so the `resolv.conf` points to the real DNS server, and WSL1 just accesses the DNS server. It doesn't really "cross" the dotted green line, cause it's already on the other side of it. Couldn't be simpler then that!

## WSL2 not on VPN

In WSL2, it has it's own network device and it's own virtual gateway adaptor (called `vEthernet (WSL)` on the Windows host) that it uses, all maintained invisibly by WSL2 magic. So the auto generated WSL2 `resolv.conf` points to the name server 172.16.48.1, and WSL2 traffic and DNS lookups is routed through 172.16.48.1

I think `ip route` typically looks like: 

```
default via 172.16.48.1 dev eth0
172.16.48.0/20 dev eth0 scope link  src 172.16.58.107
```

### When not working

For some reason the green arrow just doesn't work.... The end

Changing the MTU around _never_ helped here.

Observation: From the Windows host, you can indeed: `nslookup www.google.com 172.16.48.1` and get a response. So on the Windows Host side, 172.16.48.1 is indeed working (more on this below). So the black arrow going to the real DNS still appears to work.

### With VPN kit

So my _limited_ understanding is since we can't go through the normal network based routing, we use a pipe and cross dotted green line, VPN Kit.

The routing table is _currently_ updated to look like:

```
default via 192.168.67.1 dev eth1
172.16.48.0/20 dev eth0 scope link  src 172.16.59.104
192.168.67.0/24 dev eth1 scope link  src 192.168.67.3
```

Now, if I try to access the DNS server 172.16.48.1, then the route will still go through eth0, not eth1. And since that green arrow still is not working. Thus DNS fails, and I need the `resolv.conf` and `wsl.conf` solution.

~I got to thinking "what if I deleted the second line of the routing table?" so I did, and it still failed. As I understand it, this should be using the orange arrow, and as I stated in the "when not working" section, this actually should be working. So I'm not certain at all why this is not working at all~ Oops, I typed in a number wrong

So I got to thinking "what if I deleted the second line of the routing table?" so I did, and it does indeed work, using the orange line. And it did indeed work. My first thought was "that's an extra hop for no reason" but now my second thought is that "well, that may work more universally in ways I can't currently fathom." So I think I agree with you, if we removed that eth0 line in the routing table.

# Bonuses

1. This will close issue sakai135/wsl-vpnkit#27 (it appears alpine is just ignoring /etc/resolv.conf and I don't understand why, but I also don't care 👍)
2. The `--additional-wsl` flag will completely go away, it is no longer needed
3. No need to restart WSL, that was only wo make `wsl.conf` take affect
4. Things are simplier. Horray!

### CODE REVIEW

**Comment by andyneff** [comment]
# Summary

Made it so that all of the old ethernet device routes are remove (and restored) by the service script. The facilitates using the default resolv.conf file

- Fixed wrong line endings in env file. Now cheacking out in Windows and Linux both work as expected.
- Made README a bit easier to follow for a new comers
- Removed restart of the WSL image, since that was only needed to activate the `wsl.conf` settings.
- Since we don't need to restart the WSL image anymore, restore "start server" at the end of setup
- Added --no-start flag if that's what you want
- Removed --additional-wsl flag. That was only to setup `resolv.conf` and `wsl.conf` in other WSL images.
- Had to add onlink to default route for some weird 🤷‍♂️

**Comment by andyneff** [comment]
@AmmarRahman I had one thing I wanted to ask you about before I mark the PR ready.

The time delays here:

https://github.com/AmmarRahman/wsl-vpn/blob/ecd57bc769b781d8486b03050d612dcf244c7268/wsl-vpnkit-start.sh#L89-L95

If I understand correctly, the first one was because if you try to access a socket before socat actually starts and creates the socket, well things will clearly fail, so I put in a standard "wait for the socket to exist"

(I don't actually know exactly what `npiperelay` vs `vpnkit-tap-vsockd` do, so I don't know if they one actually needs the other started for it to start properly)

Are the other two "sleep 3" delays necessary? Did you actually have a problem for the other two? I'm wondering, do you think it's possible we could remove them all together or explain to me what they are testing so I can see if I can come up with a clever test to speed things along.

File: README.md [hunk 1]
```diff
@@ -6,24 +6,32 @@ The solution utilises [Docker's VPNKit](https://github.com/moby/vpnkit) and [Jef
...
+    - Or if you cannot (for example, always-on-VPN Corporate rules)
+        1. You can convert to image to WSL1: Windows: `wsl --set-version {WSL_NAME} 1`
+        2. Install these dependencies (e.g. `apt-get update; apt-get install socat unzip genisoinfo`)
+        3. Convert it back to WSL2: Windows: `wsl --set-version {WSL_NAME} 2`
+3. Run the setup script:
+    - *Option 1:* The preferred option since it gurantees fresh copies of all the dependencies to run the tunnel.
+        1. Install Docker Desktop on your windows machine.
+        2. Run `sudo ./wsl-vpnkit-setup.sh` from within this folder
+    - *Option 2:* If you prefer not to install Docker on your windows machine, you can run `` which would download the required files.
+        1. `sudo ./wsl-vpnkit-setup.sh --no-docker`
```
Comment (AmmarRahman) [inline README.md:22]: needs to move to the line above :)

File: README.md [hunk 2]
```diff
@@ -6,24 +6,32 @@ The solution utilises [Docker's VPNKit](https://github.com/moby/vpnkit) and [Jef
...
-### *Option 2:*
-If you prefer not to install Docker on your windows machine, you can run `sudo ./wsl-vpnkit-setup.sh --no-docker` which would download the required files.
+1. Clone the repo, in windows or WSL.
+    - If you are currently on VPN, you can only clone from WSL1 or Windows. It doesn't matter where you put the repo, as it can be removed when done.
+2. (Currently) if you are on VPN, you will have to install `socat` (and `unzip` and `isoinfo` for Option 1 below) before you can run the setup script.
+    - Easy option: Get off of VPN
+    - Or if you cannot (for example, always-on-VPN Corporate rules)
+        1. You can convert to image to WSL1: Windows: `wsl --set-version {WSL_NAME} 1`
+        2. Install these dependencies (e.g. `apt-get update; apt-get install socat unzip genisoinfo`)
+        3. Convert it back to WSL2: Windows: `wsl --set-version {WSL_NAME} 2`
```
Comment (AmmarRahman) [inline README.md:16]: I'm thinking, it might be easier if we default to unzipping on windows. What do you think?

File: wsl-vpnkit-unsetup.sh [hunk 3]
```diff
@@ -4,50 +4,23 @@ set -eu
```
Comment (AmmarRahman) [inline wsl-vpnkit-unsetup.sh:4]: I would think we can keep in the rm_if lines as they are for backwards compatibility

File: wsl-vpnkit-start.sh [hunk 4]
```diff
@@ -66,11 +86,13 @@ if [ "${EUID:-"$(id -u)"}" -ne 0 ]; then
...
+while [ ! -S "${SOCKET_PATH}" ]; do
+  sleep 0.001
+done
 vpnkit &
-sleep 3
+sleep 3 # Is this needed?
 tap &
-sleep 3
+sleep 3 # Is this needed?
 ipconfig
```
Comment (AmmarRahman) [inline wsl-vpnkit-start.sh:96]: To be honest I copied this whole file from sakai135. The second sleep can probably be omitted if we remove the ipconfig. I don't see what value that adds

**Comment by AmmarRahman** [comment]
To tell the truth, all what we are doing here is basically tunneling through windows Network interface to trick VPNs into thinking that this is windows traffic. I would imagine this to be a simpler configuration on the WSL level. 
Given my understanding of Microsoft strategy for corporate clients, I would have imagined that they would priorities finding a solution to this.  
@craigloewen-msft In our case, the use of Windows devices gives a significant experience boost to accessing corporate resources. I personally have persuaded a few engineers to switch over from Mac. However, as the experience still comes with limitations, it has been an uphill battle. 
Note that many of the users in these forums such as @andyneff or myself are power users with multiple edge cases. If you need some help with testing/bouncing ideas/feedback I would be happy to dedicate some time for that and get some of my colleagues to help. I'm pretty sure many of the other users would be happy to do the same.

File: README.md (hunk 2 above)
Comment (andyneff) [reply README.md:16]: I can totally use the same powershell trick to unzip the file (`Expand-Archive`), and after a little snooping around, I can extract files from the ISO too! (`Mount-DiskImage`, copy the files, then `Dismount-DiskImage`)! Mind if I do that as a separate PR?

So `socat` will be the only dependency.

**Comment by AmmarRahman** [comment]
❤️ This would probably address most use cases. We might address a solution for completely offline people separately
